package/endpoint_adapter/src/endpoint_adapter.c

```diff
@@ -56,15 +56,27 @@ static pk_metrics_t *MR = NULL;
 PK_METRICS_TABLE(MT, MI,
 
-  PK_METRICS_ENTRY("read/count",            "per_second",     M_U32,         M_UPDATE_COUNT,   M_RESET_DEF, read_count),
-  PK_METRICS_ENTRY("read/size/per_second",  "total",          M_U32,         M_UPDATE_SUM,     M_RESET_DEF, read_size_total),
-  PK_METRICS_ENTRY("read/size/per_second",  "average",        M_U32,         M_UPDATE_AVERAGE, M_RESET_DEF, read_size_average,
-                   M_AVERAGE_OF(MI,         read_size_total,  read_count)),
   PK_METRICS_ENTRY("error/mismatch",        "total",          M_U32,         M_UPDATE_COUNT,   M_RESET_DEF, mismatch),
-  PK_METRICS_ENTRY("write/count",           "per_second",     M_U32,         M_UPDATE_COUNT,   M_RESET_DEF, write_count),
-  PK_METRICS_ENTRY("write/size/per_second", "total",          M_U32,         M_UPDATE_SUM,     M_RESET_DEF, write_size_total),
-  PK_METRICS_ENTRY("write/size/per_second", "average",        M_U32,         M_UPDATE_AVERAGE, M_RESET_DEF, write_size_average,
-                   M_AVERAGE_OF(MI,         write_size_total, write_count))
+
+  PK_METRICS_ENTRY("rx/read/count",            "per_second",     M_U32,         M_UPDATE_COUNT,   M_RESET_DEF, rx_read_count),
+  PK_METRICS_ENTRY("rx/read/size/per_second",  "total",          M_U32,         M_UPDATE_SUM,     M_RESET_DEF, rx_read_size_total),
+  PK_METRICS_ENTRY("rx/read/size/per_second",  "average",        M_U32,         M_UPDATE_AVERAGE, M_RESET_DEF, rx_read_size_average,
+                   M_AVERAGE_OF(MI,         rx_read_size_total,  rx_read_count)),
+
+  PK_METRICS_ENTRY("tx/read/count",            "per_second",     M_U32,         M_UPDATE_COUNT,   M_RESET_DEF, tx_read_count),
+  PK_METRICS_ENTRY("tx/read/size/per_second",  "total",          M_U32,         M_UPDATE_SUM,     M_RESET_DEF, tx_read_size_total),
+  PK_METRICS_ENTRY("tx/read/size/per_second",  "average",        M_U32,         M_UPDATE_AVERAGE, M_RESET_DEF, tx_read_size_average,
+                   M_AVERAGE_OF(MI,         tx_read_size_total,  tx_read_count)),
+
+  PK_METRICS_ENTRY("rx/write/count",           "per_second",     M_U32,         M_UPDATE_COUNT,   M_RESET_DEF, rx_write_count),
+  PK_METRICS_ENTRY("rx/write/size/per_second", "total",          M_U32,         M_UPDATE_SUM,     M_RESET_DEF, rx_write_size_total),
+  PK_METRICS_ENTRY("rx/write/size/per_second", "average",        M_U32,         M_UPDATE_AVERAGE, M_RESET_DEF, rx_write_size_average,
+                   M_AVERAGE_OF(MI,         rx_write_size_total, rx_write_count)),
+
+  PK_METRICS_ENTRY("tx/write/count",           "per_second",     M_U32,         M_UPDATE_COUNT,   M_RESET_DEF, tx_write_count),
+  PK_METRICS_ENTRY("tx/write/size/per_second", "total",          M_U32,         M_UPDATE_SUM,     M_RESET_DEF, tx_write_size_total),
+  PK_METRICS_ENTRY("tx/write/size/per_second", "average",        M_U32,         M_UPDATE_AVERAGE, M_RESET_DEF, tx_write_size_average,
+                   M_AVERAGE_OF(MI,         tx_write_size_total, tx_write_count))
```

Comment:
Can anything be done to this formatting? At least github is having trouble showing it.